server/controllers/finance/reports/vouchers/index.js

```diff
@@ -107,9 +109,13 @@ function report(req, res, next) {
 
   const data = { filters };
 
-  return Vouchers.find(options)
+    return Vouchers.find(options)
     .then(rows => {
       _.extend(data, { rows });
+      return Vouchers.totalAmountByCurrency(options);
+    })
+    .then((sumAmount) => {
+      _.extend(data, { totals : sumAmount });
```

Comment:
This is much better!  Since our math is done is SQL now, it has the following advantages:
 1. It will not block the server - this code will allow other HTTP requests to be handled as fast as possible.  Since NodeJS is [single-threaded](https://stackoverflow.com/a/17959746/6251968), the previous JS function would block the event loop while it figured out what the totals were.  This would slow the system down for every user in the system.  Using MySQL is the most optimal solution.
 2. Since JS is pretty bad at math, this code is guaranteed to be correct.  You've eliminated two classes of bugs: bugs due to type incompatibilities (instead of concatenating strings, MySQL will throw an error.  JS would never tell you something went wrong) and bugs due to schema changes (JS would simply add `undefined + undefined = NaN` while SQL will throw an error). 

Just FYI, the code can be made to run faster (on multiprocessor machines) by running these two queries in parallel.  Since `Vouchers.find()` and `Vouchers.totalAmountByCurrency()` do not depend on each other, you could actually do this:
```js
return q.all([
  Vouchers.find(options),
  Vouchers.totalAmountByCurrency(options),
]).spread((rows, totals) => {

  // concurrency win!
  _.extend(data, { rows, totals});
});
```

This is a pretty advanced performance concept though and probably won't net a lot of gain on a slower machine.